src/messaging.js

```diff
@@ -252,11 +260,16 @@ Messaging.canMessageUser = function (uid, toUid, callback) {
 			}, next);
 		},
 		function (results, next) {
-			if (!results.settings.restrictChat || results.isAdmin || results.isFollowing) {
-				return next();
+			if (results.settings.restrictChat && !results.isAdmin && !results.isFollowing) {
+				return next(new Error('[[error:chat-restricted]]'));
 			}
 
-			next(new Error('[[error:chat-restricted]]'));
+			plugins.fireHook('filter:messaging.canMessageUser', {
+				uid: uid,
+				toUid: toUid,
+			}, function (err) {
+				next(err);
+			});
```

Comment:
In this case, because there is no data manipulation, I would recommend `static:messaging.canMessage`